mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -122,6 +123,83 @@ public StatisticResult addAssists(Transaction transaction, int amount) {
         return new StatisticResult(newValue);
     }
 
+    /**
+     * Adds experience to the user account. Infallible.
+     * @param transaction represents the transaction
+     * @param amount represents the amount of xp to add
+     * @return a result with the new value of xp
+     */
+    public StatisticResult addExperience(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getExperience();
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setExperience(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.EXPERIENCE);
+
+        this.updateExperience(newValue);
+
+        return new StatisticResult(newValue);
+    }
+
+    /**
+     * Adds current killstreaks to the user account. Infallible.
+     *
+     * This will increment both current and highest killstreaks.
+     *
+     * @param transaction represents the current killstreak
+     * @param amount represents the amount of killstreak to add
+     * @return a result with the new value of killstreak
+     */
+    public StatisticResult addKillstreaks(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getCurrentKillstreak();
+        int newValue = existingValue + amount;
+
+        int existingTwo = statisticsRecord.getHighestKillstreak();
+
+        if (newValue > existingTwo) {
+            int newTwo = existingTwo + amount;
+            statisticsRecord.setHighestKillstreak(newTwo);
+            this.updateHighestKillstreak(newTwo);
+        }
+
+        statisticsRecord.setCurrentKillstreak(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.HIGHEST_KILLSTREAK,KITPVP_STATISTICS.CURRENT_KILLSTREAK);
```

Comment:
You don't need to store both the highest killstreak and the current killstreak if the highest killstreak didn't change.

Reply:
Quick fix